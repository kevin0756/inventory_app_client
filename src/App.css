.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  min-height: 50%;
  padding-bottom: 10px;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

{
  box-sizing: border-box;
}

body {
  padding-top: 0;
  font-family: Helvetica Neue, Helvetica;
  background: #f7f7f7;
}

h3 {
  font-size: 18px;
  margin-top: 20px;
  margin-bottom: 5px;
}

form {
  padding: 0 40px;
}

form input {
  display: block;
  width: 100%;
  font-size: 20px;
  padding: 5px 10px;
  margin: 10px 0;
  border-radius: 5px;
  border: 1px solid #ddd;
}

form input.error {
  border-color: red;
}

form button {
  display: block;
  width: 100%;
  appearance: none;
  border-radius: 5px;
  background-color: #0D356F;
  color: #fff;
  border: none;
  font-size: 16px;
  height: 40px;
  margin-top: 30px;
}

form button:hover {
  background-color: #3B8AFB;
}

form button:disabled {
  background-color: #dbf99f;
  color: #333;
}

form button.small {
  display: inline-block;
  width: auto;
  padding: 0 20px;
  background-color: #3884F1;
  font-size: 14px;
}

form button.small:hover {
  background-color: #999;
}

.shareholder {
  display: flex;
  direction: row;
  align-items: center;
}

.shareholder button {
  margin: 0;
  margin-left: 10px;
}

table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 100%;
}

td, th {
    border: 1px solid #dddddd;
    text-align: left;
    padding: 8px;
}

tr:nth-child(even) {
    background-color: #dddddd;
}

.alignRight{
	float: right;
}

h3.alignCentre{
	text-align: center;
}