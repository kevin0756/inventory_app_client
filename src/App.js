import React, { Component } from 'react';
import {Route} from 'react-router';
import { BrowserRouter, Redirect } from 'react-router-dom'
import './App.css';
import IncorporationForm from './ProductsForm.js';
import EditProduct from './EditProductForm.js';

class HomePage extends React.Component {
	
	constructor(props) {
        super(props);
        
        this.state = {
                referrer: null,
                componentToRender: null,
                listOfProducts: [],
                product_id: null
            };
    }
	
	 deleteProduct(id) {
		 fetch('/api/inventory/delete-product-by-id/' + id, {
		      method: 'DELETE'
		    }).then(
		    		() => this.setState({listOfProducts: this.state.listOfProducts.filter((product) => product._id !== id)})
				    ,(error) => console.log(error));
	  }
	
	 componentDidMount() {
		 fetch('/api/inventory/fetch-all-products/', {
		      method: 'GET'
		    }).then(response => response.json()).then(
		    		(jsonData) => this.setState({listOfProducts: jsonData})
                    ,(error) => console.log(error));
	  }
	
	renderRedirect = (path, id) => {
		this.setState({referrer: path});
        this.setState({product_id: id});
	  }
	
	render(){
		const {referrer} = this.state;
        if (referrer) return (<BrowserRouter><div>
        <Route  exact path="/about" component={ IncorporationForm }/>
        <Route exact
          path='/edit'
          render={(props) => <EditProduct {...props} productId={this.state.product_id} />}
        />
        <Redirect to={referrer} />
        </div>
        </BrowserRouter>)
        
        var content;
        if(this.state.listOfProducts){
	        
        	content = this.state.listOfProducts.map((product, i) => 
        	       (
        	    		  <tr key={product._id.toString()}>
        	    		   <td>{product.product_name}</td>
        	    		   <td><button type="button" className="small" onClick={() => this.renderRedirect('/edit', product._id)}>edit</button></td>
        	    		   <td><button type="button" className="small" onClick={() => this.deleteProduct(product._id)}>&times;</button></td>
        	    		</tr>
        	      )
        	    );
        }
	  return (
		  
	      <div>
	      <table>
	      <thead>
	      <tr>
	      <th>Product Name</th>
	      <th>Edit</th>
	      <th>Delete</th>
	       </tr>
	      </thead>
	      <tbody>
	      { content }
	      </tbody>
	      </table>
	      <br/>
	      <button onClick={() => this.renderRedirect('/about', null)}>Add New Product</button>
	      </div>
	  );
	}
}

export default HomePage;
