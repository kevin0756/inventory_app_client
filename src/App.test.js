import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import IncorporationForm from './ProductsForm';
import EditProduct from './EditProductForm';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import  Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Component: App', () => {
it('should match its snapshot', () => {
	const tree = renderer.create(
		      <App />
		     ).toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Component: IncorporationForm', () => {
	it('should match its snapshot', () => {
		const tree = renderer.create(
			      <IncorporationForm />
			     ).toJSON();

			expect(tree).toMatchSnapshot();
	});
	
	it('should submit form values on submit', () => {
		const component = shallow(<IncorporationForm />);
	    const preventDefault = jest.fn();
	    component.find('input').simulate('change', { target: {
	        value: 'New Product' }
	  });
	    
	    component.find('form').simulate('submit', { preventDefault });
	    expect(toJson(component)).toMatchSnapshot();
	expect(preventDefault).toBeCalled();
	});

});

describe('Component: ', () => {
	it('should match its snapshot', () => {
		const tree = renderer.create(
			      <EditProduct />
			     ).toJSON();

			expect(tree).toMatchSnapshot();
	});
	
	it('it should submit form values', () => {
		const component = shallow(<EditProduct />);
	    const preventDefault = jest.fn();
	    component.find('input').simulate('change', { target: {
	        value: 'sample' }
	  });
	    
	    component.find('form').simulate('submit', { preventDefault });
	    expect(toJson(component)).toMatchSnapshot();
	expect(preventDefault).toBeCalled();
	});
});
