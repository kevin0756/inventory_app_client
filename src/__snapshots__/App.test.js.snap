// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  it should submit form values 1`] = `
<form
  onSubmit={[Function]}
>
  <h3
    className="alignCentre"
  />
  <h4>
    Edit Product Details
  </h4>
  <div
    className="shareholder"
  >
    <input
      onChange={[Function]}
      type="text"
      value="sample"
    />
  </div>
  <button
    className="small alignRight"
    onClick={[Function]}
  >
    Back
  </button>
  <button>
    Submit
  </button>
</form>
`;

exports[`Component:  should match its snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h3
    className="alignCentre"
  >
    
  </h3>
  <h4>
    Edit Product Details
  </h4>
  <div
    className="shareholder"
  >
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <button
    className="small alignRight"
    onClick={[Function]}
  >
    Back
  </button>
  <button>
    Submit
  </button>
</form>
`;

exports[`Component: App should match its snapshot 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Product Name
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <br />
  <button
    onClick={[Function]}
  >
    Add New Product
  </button>
</div>
`;

exports[`Component: EditProduct should match its snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h3
    className="alignCentre"
  >
    
  </h3>
  <h4>
    Edit Product Details
  </h4>
  <div
    className="shareholder"
  >
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <button
    className="small alignRight"
    onClick={[Function]}
  >
    Back
  </button>
  <button>
    Submit
  </button>
</form>
`;

exports[`Component: IncorporationForm should match its snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h3
    className="alignCentre"
  >
    Data Submitted Successfully
  </h3>
  <h4>
    Products
  </h4>
  <div
    className="shareholder"
  >
    <input
      onChange={[Function]}
      placeholder="Product #1"
      type="text"
      value=""
    />
    <button
      className="small"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <button
    className="small"
    onClick={[Function]}
    type="button"
  >
    Add Product
  </button>
  <button
    className="small alignRight"
    onClick={[Function]}
  >
    Back
  </button>
  <button>
    Submit
  </button>
</form>
`;

exports[`Component: IncorporationForm should submit form values on submit 1`] = `
<form
  onSubmit={[Function]}
>
  <h3
    className="alignCentre"
  >
    Data Submitted Successfully
  </h3>
  <h4>
    Products
  </h4>
  <div
    className="shareholder"
  >
    <input
      onChange={[Function]}
      placeholder="Product #1"
      type="text"
      value="New Product"
    />
    <button
      className="small"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <button
    className="small"
    onClick={[Function]}
    type="button"
  >
    Add Product
  </button>
  <button
    className="small alignRight"
    onClick={[Function]}
  >
    Back
  </button>
  <button>
    Submit
  </button>
</form>
`;
